src/julia_internal.h

```diff
@@ -127,11 +127,17 @@ JL_DLLEXPORT extern const char *jl_filename;
 JL_DLLEXPORT jl_value_t *jl_gc_pool_alloc(jl_ptls_t ptls, int pool_offset,
                                           int osize);
 JL_DLLEXPORT jl_value_t *jl_gc_big_alloc(jl_ptls_t ptls, size_t allocsz);
-int jl_gc_classify_pools(size_t sz, int *osize);
+int jl_gc_classify_pools(size_t sz, size_t alignment, int *osize);
 extern jl_mutex_t gc_perm_lock;
 void *jl_gc_perm_alloc_nolock(size_t sz, int zero, unsigned align, unsigned offset);
 void *jl_gc_perm_alloc(size_t sz, int zero, unsigned align, unsigned offset);
 
+#define JL_SMALL_BYTE_ALIGNMENT 16
+#define JL_CACHE_BYTE_ALIGNMENT 64
+// JL_HEAP_ALIGNMENT is the maximum alignment that the GC can provide
+#define JL_HEAP_ALIGNMENT JL_SMALL_BYTE_ALIGNMENT
+#define GC_MAX_SZCLASS (2032-sizeof(void*))
```

Comment:
I'm not sure what the `sizeof(void*)` comes from, but this PR is supposed to be changing this list (https://github.com/JuliaLang/julia/pull/21959#discussion_r117473120)